platform/src/main/java/org/hillview/storage/GenericLogs.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2018 MapleLabs. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.storage;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Schema;
+import org.hillview.table.Table;
+import org.hillview.table.api.ContentsKind;
+import org.hillview.table.api.ITable;
+import org.hillview.storage.FileSetDescription;
+import org.hillview.utils.Utilities;
+
+import java.util.Map;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.File;
+
+import io.krakens.grok.api.*;
+
+/**
+ * Reads Generic logs into ITable objects.
+ */
+public class GenericLogs {
+    private static String hostName;
+    private Schema schema;
+    private Grok grok;
+    private String logFormat;
+
+    GenericLogs(String logFormat) {
+        this.logFormat = logFormat;
+        GrokCompiler grokCompiler = GrokCompiler.newInstance();
+        grokCompiler.registerDefaultPatterns();
+        grokCompiler.registerPatternFromClasspath("/patterns/log-patterns");
+        this.grok = grokCompiler.compile(this.logFormat, true);
+        schema = new Schema();
+    }
+
+    static {
+        GenericLogs.hostName = Utilities.getHostName();
+    }
+
+    public static class LogFileLoader extends TextFileLoader {
+        private static GenericLogs genLog;
```

Comment:
I think this one has to be non-static too.
Alternatively, you can make the LogFileLoader to be a non-static inner class of GenericLogs.

Reply:
@mbudiu-vmw Done.